### ISSUE TITLE
Reproducibility Question for PCL-TDGCN (SEED Subject-Independent)

### ISSUE THREAD

**Issue by IiCEight** [YYingDL/PCL-TDGCN#2]
Thank you for the great work and for making the code publicly available. I really appreciate the effort put into this paper and the clean implementation. I am trying to reproduce the results on the SEED dataset and ran into a gap I cannot explain — I would be very grateful for any guidance.

 Environment

  - PyTorch 2.x, CUDA, RTX 4090
  - Python 3.12
  - SEED dataset, **standard de_LDS features (ExtractedFeatures/)**

  Problem

  I use the official implementation with only modifying the seed data path and leave everything unchanged. But I consistently get ~87% accuracy on SEED subject-independent, while Table IV reports 96.77% ± 6.83%. **I keep every argument as default**.

```
Cross-Subject Validation Complete
Mean Accuracy: 0.8709 ± 0.0824
Individual Subject Accuracies: [0.88833235 0.88096641 0.8777254  0.91985857 0.77695934 0.75132587
 0.87743076 0.77401296 0.8087802  0.87684148 0.98556276 1.
 0.89776075 1.         0.74749558]
Aggregated Confusion Matrix:
[[ 9959  6355   486]
 [ 4280 11540   740]
 [ 1200  2621 13729]]

```
  
  What I verified

  - Model code uses this repo exactly (only `get_allmats_name` in `utils.py` for matching my SEED folder structure is modified)
  - Data loading uses de_LDS keys of `x.mat` in `SEED/ExtractedFeatures/`
  - Training script matches train_independent.py exactly
  - --session 0, --epochs 1000, --batch_size 48, all default hyperparameters

  Questions

  1. Is there any preprocessing beyond the standard public de_LDS features?
  2. Are there any undisclosed hyperparameters or training tricks needed to reproduce 96.77%?

Thank you again for your time. Any help or clarification would be greatly appreciated!

**Comment by YYingDL** [comment]
Dear all,

I checked the code and revised line 340 as follows:

python
# Domain adversarial loss
global_transfer_loss = dann_loss(
    src_feature + 0.005 * torch.randn_like(src_feature).to(args.device),
    tar_feature + 0.005 * torch.randn_like(tar_feature).to(args.device),
    src_prob, tar_output_cls
)

**Comment by IiCEight** [comment]
Thank you very much for getting back to me so quickly and for providing the code revision for line 340.

I tested the correction of the `tar_output_cls`. While it unfortunately does not resolve the performance gap.

The current implementation of the `DAANLoss` module only calculates and returns the `global_loss` (marginal distribution alignment) and completely ignores `source_logits` and `target_logits`. So the last two parameters (`src_prob, tar_output_cls`) are never actually used in the calculation.

```python
  class DAANLoss(nn.Module):
      # ...
      def forward(self, source, target, source_logits, target_logits):
          global_loss = self.get_global_adversarial_result(source, target)

          # The following critical DAAN components are commented out/missing:
          # self.d_g = self.d_g + 2 * (1 - 2 * global_loss.cpu().item())
          # self.d_l = self.d_l + 2 * (1 - 2 * (local_loss / self.num_class).cpu().item())
          # adv_loss = (1 - self.dynamic_factor) * global_loss + self.dynamic_factor * local_loss

          return global_loss
```

Could you kindly check if the complete implementation—was perhaps accidentally omitted something from this public release?

Thank you again for your time, patience, and dedication to maintaining this repository. I really appreciate your help!

**Comment by YYingDL** [comment]
Dear IiCEight,

I only use the global loss. Then, I ran the program again and found no problems. 

I suggest you refer to this solution from another project.  https://github.com/YYingDL/NSAL-DGAT/issues/1
我将SEED数据集的ExtractedFeatures文件夹下的45个.mat文件通过复制粘贴分为三个session存储在同一个文件夹下，每个文件夹是15个.mat文件，具体结构如下：
--mainfile
----1
----2
----3
再把mainfile的路径传入utils.py的seed3里面，运行train.py即可

**Comment by IiCEight** [comment]
Dear YYingDL,
I finally found the problem. In your current setting, the `eval_interval = 10`  is applied which will lead to calculate the test acc every 10 epochs. That's why it will get the 87% acc. 
Once I set it as `eval_interval = 1`, your execllent model produces a pretty reasonable result which the acc is **93%**.
Thanks for your time, wish you have a good day!
